src/cli/cli.ts

```diff
@@ -0,0 +1,708 @@
+#!/usr/bin/env node
+
+/**
+ * Copyright (C) 2022-2024 Permanent Data Solutions, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+// eslint-disable-next-line header/header -- This is a CLI file
+import { program } from 'commander';
+
+import { intentsUsingYears, isValidIntent, validIntents } from '../types/io.js';
+import { mIOToken } from '../types/token.js';
+import { version } from '../version.js';
+import { joinNetwork } from './commands/joinNetwork.js';
+import { transfer } from './commands/transfer.js';
+import {
+  addressOptions,
+  arNSAuctionPricesOptions,
+  epochOptions,
+  getVaultOptions,
+  globalOptions,
+  initiatorAndNameOptions,
+  joinNetworkOptions,
+  nameOptions,
+  optionMap,
+  paginationAddressOptions,
+  paginationOptions,
+  tokenCostOptions,
+  transferOptions,
+} from './options.js';
+import {
+  AddressOptions,
+  AuctionPricesOptions,
+  EpochOptions,
+  GetTokenCostOptions,
+  GetVaultOptions,
+  GlobalOptions,
+  InitiatorOptions,
+  NameOptions,
+  PaginationAddressOptions,
+  PaginationOptions,
+} from './types.js';
+import {
+  epochInputFromOptions,
+  formatIOWithCommas,
+  makeCommand,
+  paginationParamsFromOptions,
+  readIOFromOptions,
+  requiredAddressFromOptions,
+  requiredInitiatorFromOptions,
+  requiredNameFromOptions,
+  runCommand,
+} from './utils.js';
+
+makeCommand({
+  name: 'ar.io', // TODO: can it be ar.io?
+  description: 'AR.IO Network CLI',
+  options: globalOptions,
+})
+  .version(version)
+  .helpCommand(true);
+
+makeCommand({
+  name: 'info',
+  description: 'Get network info',
+  options: [],
+}).action(async (_, command) => {
+  await runCommand<GlobalOptions>(command, async (options) => {
+    return readIOFromOptions(options).getInfo();
+  });
+});
```

Comment:
could probably condense these in the makeCommand API

```suggestion
makeCommand<GlobalOptions>({
  name: 'info',
  description: 'Get network info',
  action: (options) => readIOFromOptions(options).getInfo();
})
```

Reply:
and makeCommand could have `T = GlobalOptions` default